mp/src/game/shared/momentum/mom_stickybomb.cpp

```diff
@@ -0,0 +1,366 @@
+
+        const auto pWepScript = g_pWeaponDef->GetWeaponScript(WEAPON_STICKYLAUNCHER);
+
+        const char *pParticle = pWepScript->pKVWeaponParticles->GetString("StickybombTrail_TF2");
+
+        ParticleProp()->Create(pParticle, PATTACH_ABSORIGIN_FOLLOW);
+        m_bPulsed = false;
+
+        CMomentumStickybombLauncher *pLauncher = dynamic_cast<CMomentumStickybombLauncher *>(m_hLauncher.Get());
+
+        if (pLauncher)
+        {
+            pLauncher->AddStickybomb(this);
+        }
+
+        // Now stick our initial velocity into the interpolation history
+        CInterpolatedVar<Vector> &interpolator = GetOriginInterpolator();
+        interpolator.ClearHistory();
+
+        CInterpolatedVar<QAngle> &rotInterpolator = GetRotationInterpolator();
+        rotInterpolator.ClearHistory();
+
+        float changeTime = GetLastChangeTime(LATCH_SIMULATION_VAR);
+
+        // Add a sample 1 second back.
+        Vector vCurOrigin = GetLocalOrigin() - m_vInitialVelocity;
+        interpolator.AddToHead(changeTime - 1.0f, &vCurOrigin, false);
+
+        QAngle vCurAngles = GetLocalAngles();
+        rotInterpolator.AddToHead(changeTime - 1.0f, &vCurAngles, false);
+
+        // Add the current sample.
+        vCurOrigin = GetLocalOrigin();
+        interpolator.AddToHead(changeTime, &vCurOrigin, false);
+
+        rotInterpolator.AddToHead(changeTime, &vCurAngles, false);
+    }
+}
+
+int CMomStickybomb::DrawModel(int flags)
+{
+    // Don't draw the stickybomb during the time defined by the cvar.
+    if (gpGlobals->curtime - m_flSpawnTime < mom_sj_stickybomb_drawdelay.GetFloat())
+    {
+        return 0;
+    }
+
+    return BaseClass::DrawModel(flags);
+}
+
+void CMomStickybomb::Simulate()
+{
+    if (!m_bPulsed)
+    {
+        if ((gpGlobals->curtime - m_flCreationTime) >= 0.8f)
+        {
+            const auto pWepScript = g_pWeaponDef->GetWeaponScript(WEAPON_STICKYLAUNCHER);
+
+            const char *pParticle = pWepScript->pKVWeaponParticles->GetString("StickybombPulse_TF2");
+
+            ParticleProp()->Create(pParticle, PATTACH_ABSORIGIN_FOLLOW);
+            m_bPulsed = true;
+        }
+    }
+    BaseClass::Simulate();
+}
+
+#else
+
+void CMomStickybomb::SetupInitialTransmittedGrenadeVelocity(const Vector &velocity) { m_vInitialVelocity = velocity; }
+
+CMomStickybomb *CMomStickybomb::Create(const Vector &position, const QAngle &angles, const Vector &velocity,
+                                       const AngularImpulse &angVelocity, CBaseCombatCharacter *pOwner)
+{
+    CMomStickybomb *pStickybomb =
+        static_cast<CMomStickybomb *>(CBaseEntity::CreateNoSpawn("momentum_stickybomb", position, angles, pOwner));
+
+    if (pStickybomb)
+    {
+        DispatchSpawn(pStickybomb);
+        pStickybomb->InitStickybomb(velocity, angVelocity, pOwner);
+        pStickybomb->ApplyLocalAngularVelocityImpulse(angVelocity);
+        pStickybomb->SetAbsVelocity(velocity);
+        pStickybomb->SetOwnerEntity(pOwner);
```

Comment:
No need to set owner here, it's automatically done in the CreateNoSpawn call above